common/src/main/java/com/revenuecat/purchases/common/verification/SigningManager.kt

```diff
@@ -0,0 +1,60 @@
+package com.revenuecat.purchases.common.verification
+
+import android.util.Base64
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.networking.HTTPResult
+import com.revenuecat.purchases.strings.NetworkStrings
+import java.security.SecureRandom
+
+class SigningManager(
+    private val signatureVerifier: SignatureVerifier
+) {
+    private companion object {
+        const val NONCE_BYTES_SIZE = 12
+        const val SALT_BYTES_SIZE = 16
+    }
+
+    fun createRandomNonce(): String {
+        val bytes = ByteArray(NONCE_BYTES_SIZE)
+        SecureRandom().nextBytes(bytes)
+        return String(Base64.encode(bytes, Base64.DEFAULT))
+    }
+
+    @Suppress("LongParameterList", "ReturnCount")
+    fun verifyResponse(
+        requestPath: String,
+        signature: String?,
+        nonce: String,
+        body: String?,
+        requestTime: String?,
+        eTag: String?
+    ): HTTPResult.VerificationStatus {
+        if (signature == null) {
+            errorLog(NetworkStrings.VERIFICATION_MISSING_SIGNATURE.format(requestPath))
+            return HTTPResult.VerificationStatus.ERROR
+        }
+        if (requestTime == null) {
+            errorLog(NetworkStrings.VERIFICATION_MISSING_REQUEST_TIME.format(requestPath))
+            return HTTPResult.VerificationStatus.ERROR
+        }
+
+        // No body means NOT_MODIFIED response. We verify the etag instead.
```

Comment:
I want to confirm this logic is correct. This was extracted from iOS.

Reply:
Yeah that's correct. I took that from Khepri, and validated that with real signatures.
FWIW though, in iOS and the backend we determine what to sign based on the status code though, not just the presence of a body.

Reply:
That's a good point... I will mimic that logic here instead of relying on the body existence

Reply:
I changed to this logic in https://github.com/RevenueCat/purchases-android/pull/826